integration-tests/manifests/annotations/validate_annotations_namespace_test.go

```diff
@@ -127,3 +132,300 @@ func TestPythonOnlyNamespace(t *testing.T) {
+			StatefulSets: []string{""},
+		},
+		Python: auto.AnnotationResources{
+			Namespaces:   []string{""},
+			DaemonSets:   []string{""},
+			Deployments:  []string{""},
+			StatefulSets: []string{""},
+		},
+	}
+	jsonStr, err := json.Marshal(annotationConfig)
+	if err != nil {
+		t.Error("Error:", err)
+	}
+
+	startTime := time.Now()
+	updateTheOperator(t, clientSet, string(jsonStr))
+	if err != nil {
+		t.Errorf("Failed to get deployment app: %s", err.Error())
+	}
+
+	deployment, err := clientSet.AppsV1().Deployments(uniqueNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		fmt.Printf("Error retrieving deployment: %v\n", err)
+		os.Exit(1)
+	}
+
+	//Removing all annotations
+	deployment.ObjectMeta.Annotations = nil
+	deployment, err = clientSet.AppsV1().Deployments(uniqueNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	_, err = clientSet.AppsV1().Deployments(uniqueNamespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+	if err != nil {
+		fmt.Printf("Error updating deployment: %v\n", err)
+		os.Exit(1)
+	}
+
+	err = util.WaitForNewPodCreation(clientSet, deployment, startTime)
+	if err != nil {
+		fmt.Printf("Error waiting for pod creation: %v\n", err)
+		os.Exit(1)
+	}
+
+	deploymentPods, err := clientSet.CoreV1().Pods(uniqueNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error listing pods: %v\n", err)
+		os.Exit(1)
+	}
+	//Check if operator has added back the annotations
+	checkIfAnnotationExists(clientSet, deploymentPods, []string{injectJavaAnnotation, autoAnnotateJavaAnnotation})
+
+}
+
+// Creating two apps - First app is annotated
+// Second app is not annotated but on the same namespace as the first app
+// Annotate the namespace of the apps and make sure only the non annotated app was restarted
+// Also tests if a resource is manually annotated and now its namespace is added for auto annotation
+// the resource should not be modified and should not be restarted (auto-annotation annotation does not exist)
+func TestOnlyNonAnnotatedAppsShouldBeRestarted(t *testing.T) {
+
+	clientSet := setupTest(t)
+	randomNumber, err := rand.Int(rand.Reader, big.NewInt(9000))
+	if err != nil {
+		panic(err)
+	}
+	randomNumber.Add(randomNumber, big.NewInt(1000)) //adding a hash to namespace
+	uniqueNamespace := fmt.Sprintf("multiple-resources-%d", randomNumber)
+	if err := createNamespaceAndApplyResources(t, clientSet, uniqueNamespace, []string{sampleDeploymentYamlNameRelPath}); err != nil {
+		t.Fatalf("Failed to create/apply resoures on namespace: %v", err)
+	}
+	defer func() {
+		if err := deleteNamespaceAndResources(clientSet, uniqueNamespace, []string{sampleDeploymentYamlNameRelPath}); err != nil {
+			t.Fatalf("Failed to delete namespaces/resources: %v", err)
+		}
+	}()
+	if err := createNamespaceAndApplyResources(t, clientSet, uniqueNamespace, []string{sampleNginxAppYamlNameRelPath}); err != nil {
```

Comment:
Could this be created as part of the first `createNamespaceAndApplyResources` in the slice parameter?

Reply:
yes, I will make them into one function